perun-core/src/main/java/cz/metacentrum/perun/core/impl/ExtSourceSql.java

```diff
@@ -119,299 +125,153 @@ public ExtSourceSql() {
 		return this.querySource(query, null, 0);
 	}
 
-	protected List<Map<String,String>> querySource(String query, String searchString, int maxResults) throws InternalErrorException {
-		log.debug("Searching for '{}' in external source 'url:{}'", searchString, getAttributes().get("url"));
-
-		this.checkAndSetPrerequisites();
-
-		try (PreparedStatement st = getPreparedStatement(query, searchString, maxResults)) {
-			try (ResultSet rs = st.executeQuery()) {
-				List<Map<String, String>> subjects = new ArrayList<>();
-
-				log.trace("Query {}", query);
+	/**
+	 * Helper structure for holding data about database column mappings.
+	 */
+	private static class ColumnMapping {
+		String columnName;
+		int columnIndex;
+		String attributeName;
+		boolean blob;
+
+		public ColumnMapping(String columnName, int columnIndex, String attributeName, boolean blob) {
+			this.columnName = columnName;
+			this.columnIndex = columnIndex;
+			this.attributeName = attributeName;
+			this.blob = blob;
+		}
+	}
 
-				while (rs.next()) {
-					Map<String, String> map = new HashMap<>();
+	private static final Set<String> BASE_COLUMNS = ImmutableSet.of("firstName", "lastName", "middleName", "titleBefore", "titleAfter", "login");
 
-					try {
-						map.put("firstName", rs.getString("firstName"));
-					} catch (SQLException e) {
-						// If the column doesn't exists, ignore it
-						map.put("firstName", null);
-					}
-					try {
-						map.put("lastName", rs.getString("lastName"));
-					} catch (SQLException e) {
-						// If the column doesn't exists, ignore it
-						map.put("lastName", null);
-					}
-					try {
-						map.put("middleName", rs.getString("middleName"));
-					} catch (SQLException e) {
-						// If the column doesn't exists, ignore it
-						map.put("middleName", null);
-					}
-					try {
-						map.put("titleBefore", rs.getString("titleBefore"));
-					} catch (SQLException e) {
-						// If the column doesn't exists, ignore it
-						map.put("titleBefore", null);
-					}
-					try {
-						map.put("titleAfter", rs.getString("titleAfter"));
-					} catch (SQLException e) {
-						// If the column doesn't exists, ignore it
-						map.put("titleAfter", null);
-					}
-					try {
-						map.put("login", rs.getString("login"));
-					} catch (SQLException e) {
-						// If the column doesn't exists, ignore it
-						map.put("login", null);
+	protected List<Map<String,String>> querySource(String query, String searchString, int maxResults) throws InternalErrorException {
+		log.debug("Searching for '{}' in external source '{}'", searchString, getName());
+		try (Connection con = dataSource.getConnection()) {
+			try (PreparedStatement st = con.prepareStatement(query)) {
+				// Substitute the ? in the query by the searchString
+				if (searchString != null && !searchString.isEmpty()) {
```

Comment:
We can use ``StringUtils.isNotEmpty(searchString)`` or even ``StringUtils.isNotBlank(searchString)``. Its not necessary, I just like those methods, since they can make code more readable if we check multiple conditions.

Reply:
OK, this code was there before my change, replaced.